### DISCUSSION TITLE
Configurable interlock delay

### DISCUSSION THREAD

**Discussion by t-8ch** [arendst/Tasmota#11407] [Ideas]
Currently Tasmota always introduces a 50 milliseconds delay when switching interlocked channels.

See https://github.com/arendst/Tasmota/issues/6313#issuecomment-526161676
https://github.com/arendst/Tasmota/blob/c87a4df7e982506c4cb242fea39b8b5f3bc6ee51/tasmota/support_tasmota.ino#L565

I have a device that requires at least 500 milliseconds of delay as per the manufacturer:

    When changing operating direction, the changeover time must
    be at least 0.5 seconds. The switches used must not carry out
    simultaneous UP / DOWN commands. The motor must only be
    used for the purposes described in the instructions.

( https://www.jarolift.de/files/jarolift/manuals/motoren/en/Jarolift_TDEP_EN.pdf )

It would be great if Tasmota could provide a command to change the default interlock delay (`InterlockDelay`)
If this is something that would be accepted I am offering to provide the code.

Side note:
Currently the delay is executed for each device being turned off instead of once after all devices have been turned off.
This means that the delay will grow with the size of the interlock group.
Is this intentional?
(PR submitted in https://github.com/arendst/Tasmota/pull/11410 )

**Comment by ntninja** [comment]
There is still no way to configure the interlock delay, correct?

Schlotterer blind devices also require a 500ms interlock delay:
https://id.schlotterer.com/p/71005422023a (page 5, section 5.3/*Important*)

**Reply by ntninja** [reply]
Ok, it’s not directly available for interlock delay but it *is* available for shutter control and defaults to 500ms there: https://tasmota.github.io/docs/Blinds-and-Shutters/#motor-stop-time

So at least for blinds this is a solved issue at least. :grin:

**Reply by sfromis** [reply]
There is a hardcoded delay of 50 ms, without a configuration option. It would be simple to change that into a `#define symbol` defaulting to 50. Same approach is used with many other constants.
See this PR for where it happens: https://github.com/arendst/Tasmota/pull/11410
